Cabal/Distribution/Simple/PreProcess.hs

```diff
@@ -382,7 +382,7 @@ ppHsc2hs bi lbi =
     platformIndependent = False,
     runPreProcessor = mkSimplePreProcessor $ \inFile outFile verbosity -> do
       (gccProg, _) <- requireProgram verbosity gccProgram (withPrograms lbi)
-      rawSystemProgramConf verbosity hsc2hsProgram (withPrograms lbi) $
+      rawSystemProgramConf verbosity hsc2hsProgram (withPrograms lbi) . nub $
```

Comment:
1. Please use `ordNubRight` instead of `nub` (lives in `D.S.Utils`)
1. Please add a comment with a link to this PR.